### PR TITLE
[21982] Tests for regression test: fix sequence of alias of primitives inside modules

### PR DESCRIPTION
**Pull request by richiware**
## Description
This  PR adds tests for the new type in `dds-types-test` repository which was included as regression test for `fastddsgen`.

Depends on:
- eprosima/fast-dds-gen#415

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- *N/A* Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- *N/A* Changes are API compatible. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp [hunk 1]
```diff
@@ -1434,6 +1441,76 @@ TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_BoundedBigSequences)
...
+    alias_descriptor->name(alias_my_aliasstring_name);
+    alias_descriptor->base_type(alias_type);
+    alias_type = DynamicTypeBuilderFactory::get_instance()->create_type(alias_descriptor)->build();
+
+    alias_descriptor = traits<TypeDescriptor>::make_shared();
+    alias_descriptor->kind(TK_ALIAS);
+    alias_descriptor->name(alias_my_sequencestring_name);
+    alias_descriptor->base_type(DynamicTypeBuilderFactory::get_instance()->create_sequence_type(alias_type,
+            LENGTH_UNLIMITED)->build());
+    alias_type = DynamicTypeBuilderFactory::get_instance()->create_type(alias_descriptor)->build();
```
Comment (Mario-DL) [inline test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp:1470]: Same here

File: test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp [hunk 2]
```diff
@@ -1434,6 +1441,76 @@ TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_BoundedBigSequences)
     EXPECT_EQ(DynamicDataFactory::get_instance()->delete_data(data), RETCODE_OK);
 }
 
+TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_My_Structure)
+{
+    TypeDescriptor::_ref_type alias_descriptor {traits<TypeDescriptor>::make_shared()};
+    alias_descriptor->kind(TK_ALIAS);
+    alias_descriptor->name(alias_my_char_name);
+    alias_descriptor->base_type(DynamicTypeBuilderFactory::get_instance()->get_primitive_type(TK_CHAR8));
+    DynamicType::_ref_type alias_type {DynamicTypeBuilderFactory::get_instance()->create_type(alias_descriptor)->build()};
```
[the remark this answers is not in the document]
Comment (richiware) [reply test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp]: Done in 463d44a1b

File: test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp [hunk 3]
```diff
@@ -1434,6 +1441,76 @@ TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_BoundedBigSequences)
...
+
+    alias_descriptor = traits<TypeDescriptor>::make_shared();
+    alias_descriptor->kind(TK_ALIAS);
+    alias_descriptor->name(alias_my_string_name);
+    alias_descriptor->base_type(DynamicTypeBuilderFactory::get_instance()->create_sequence_type(alias_type,
+            20)->build());
+    alias_type = DynamicTypeBuilderFactory::get_instance()->create_type(alias_descriptor)->build();
```
[the remark this answers is not in the document]
Comment (richiware) [reply test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp]: Done in 463d44a1b

File: test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp [hunk 4]
```diff
@@ -1434,6 +1441,76 @@ TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_BoundedBigSequences)
...
+
+    alias_descriptor = traits<TypeDescriptor>::make_shared();
+    alias_descriptor->kind(TK_ALIAS);
+    alias_descriptor->name(alias_my_aliasstring_name);
+    alias_descriptor->base_type(alias_type);
+    alias_type = DynamicTypeBuilderFactory::get_instance()->create_type(alias_descriptor)->build();
```
[the remark this answers is not in the document]
Comment (richiware) [reply test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp]: Done in 463d44a1b

File: test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp (hunk 1 above)
Comment (richiware) [reply test/feature/dynamic_types/dds_types_tests/DynamicTypesSequencesDDSTypesTests.cpp]: Done in 463d44a1b

**Review by Mario-DL — APPROVED** [review]
LGTM with Green CI

**Pull request merged**